Instant Search: Hide search options separators from screen readers.

a8ck3n opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Moves the separator "•" from CSS into the DOM and sets the `aria-hidden` attribute so that the separators are ignored by screen readers.

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

- Removes the CSS that added the "•" separators.
- Adds them inline to the HTML.
- Updated CSS to maintain styling.
- Adds `aria-hidden` attribute to hide from screen readers.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

1. Load the Instant Search overlay.
2. Inspect the "•" separators.
3. Confirm they all have `aria-hidden="true"` set.
4. Test in Firefox using the Accessibility inspector.
5. Confirm the separators are not visible there.

Screenshot:

![NoSeps](https://user-images.githubusercontent.com/40267301/180377673-1ce58c1d-327c-4bf6-9978-378494c97141.png)

a8ck3n committed:
[not verified] Instant Search: Fix keyboard handling on search options.

Addresses the Relevance, Newest, and Oldest links. Now using HTML buttons instead of anchors with some minor CSS tweaks to keep them looking pretty. As buttons, they get the proper default keyboard handling and are recognized in the tab-loop more consistently.

a8ck3n committed:
Instant Search: Fix keyboard handling on search options.

Addresses the Relevance, Newest, and Oldest links. Now using HTML buttons instead of anchors with some minor CSS tweaks to keep them looking pretty. As buttons, they get the proper default keyboard handling and are recognized in the tab-loop more consistently.

a8ck3n committed:
Add changelog file.

a8ck3n committed:
[not verified] Add changelog file.

a8ck3n committed:
Move separators from CSS to HTML.

Allows us to apply ARIA attributes and hide the separators from screen readers.

a8ck3n committed:
[not verified] Move separators from CSS to HTML.

Allows us to apply ARIA attributes and hide the separators from screen readers.

a8ck3n committed:
Merge branch 'trunk' into fix/seach-options-separators

annacmc committed:
Search: Consistent design for focus states in Search overlay (#25260)

chriskmnds committed:
Admin menu: Add add-ons menu item (#25085)

Co-authored-by: Karen Attfield <karenlattfield@gmail.com>

retrofox committed:
Jetpack: add a transform from/to core/video to VPBlock (#25154)

Co-authored-by: Samiff <samiff@users.noreply.github.com>
Co-authored-by: Karen Attfield <karenlattfield@gmail.com>

danielpost committed:
Jetpack Social: Add posts box (#25166)

SiobhyB committed:
[RNMobile] Block Inserter: Limit Jetpack block collection to the WordPress app (#25155)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

annacmc reviewed (dismissed):
works as expected for me! tested out in chrome, safari & firefox, and the firefox accessibility inspector 👍

Commit:
[Package] Jetpack Backup: add review request component (#25161)

bisko committed:
Fix/dedicated sync enable callables for dedicated sync requests (#25143)

* Dedicated Sync: Enable sending callables from non-admin requests if Dedicated Sync is enabled

* changelog

* Dedicated Sync: Initialize listener on dedicated sync requests

Co-authored-by: Karen Attfield <karenlattfield@gmail.com>
Co-authored-by: Foteini Giannaropoulou <giannaropoulou.foteini@gmail.com>

coder-karen committed:
Jetpack 11.2-beta changelog (#25271)

Co-authored-by: Steve D. <33553323+sdixon194@users.noreply.github.com>

coder-karen committed:
Jetpack: init 11.3 cycle (#25274)

jeherve committed:
Repo Gardening: update documentation (#25275)

jeherve committed:
Repo Gardening: create a new "Reply to customers Reminder" task (#25170)

dhasilva committed:
My Jetpack: Add button to ProductCard with absent plugin (#25204)

* [not verified] add button when plugin is absent in ProductCard

The component uses a simple span element in this case

* [not verified] changelog

* [not verified] fix the button variant

* [not verified] bump my-jetpack version to 1.8.2-alpha

coder-karen committed:
Jetpack 11.2 beta changelog and readme editorial (#25276)

Co-authored-by: Steve D. <33553323+sdixon194@users.noreply.github.com>

anomiex committed:
renovate: Use configured version of PHP when running composer (#25112)

In the absence of configuration or certain dependencies, renovate
currently defaults to running composer with PHP 7.4. Or, more
specifically, it defaults to whatever is the 'latest' tag on the
`renovate/php` docker package.

We do everything else with PHP 8.0, so configure renovate to use that
too.

jeherve committed:
SEO Tools: avoid conflicts with SEOPress and SEOKEY (#25277)

Co-authored-by: Samiff <samiff@users.noreply.github.com>

yansern committed:
Subscriptions Block: Fix subscriber count incorrect position when padding is enabled. (#25262)

* Use flex & text-align to position subscription count. Use one html structure for subscription count.

* Add changelog.

* Update tests.

* Update fix-subscriber-count-padding

Co-authored-by: Samiff <samiff@users.noreply.github.com>

matticbot committed:
Lock file maintenance (#25279)

Composer lock files only this time.

Co-authored-by: Renovate Bot <bot@renovateapp.com>

annacmc committed:
Search: Record Meter, remove feature flag for record meter (#25061)

Co-authored-by: Chris R <chris.rosser@automattic.com>

a8ck3n committed:
Merge branch 'fix/seach-options-separators' of github.com:Automattic/jetpack into fix/seach-options-separators

a8ck3n commented:
So much for rebasing via the GH Desktop app…

a8ck3n commented:
Closed in favour of #25283.
